### PR TITLE
AlertRules: Update design to align it with recent changes to lists and tables 

### PR DESCRIPTION
**Pull request by torkelo**
Part of an effort to align designs and make Grafana more consistent and polished. We have done similar changes to other dense lists and tables. Making the design less boxy, clean and simpler 

This now aligns the Alert rule list to be more similar to other lists and tables. 

* Remove background for header
* Align border to use colors.borders.weak which is the one we use in most places (weak is a bit too weak in the dark theme and we will fix this before G10 as this is an issue for Scenes so we need to fix this anyway)
* Fix some alignment issues (toggle and state tag where not vertically centered) 
* Reduce some spacing that are no longer needed 

The inner table (DynamicTable) still has background for header and striped backgrounds. The striped background is an issue for the forms that are place on the even rows (they look kind of bad as the form styles do not work well on secondary background). But I don't want to do too big a change at once so leaving the DynamicTable styles mostly intact in this PR. 

Before:
Screenshot from 2023-04-13 14-06-49

After:
Screenshot from 2023-04-13 14-06-24

### CODE REVIEW

**Comment by torkelo** [comment]
For context here is the dashboard list
Screenshot from 2023-04-13 14-13-15

Table: 
Screenshot from 2023-04-13 14-13-54

Interactive tab (very simlar to alert rules):
Screenshot from 2023-04-13 11-31-20

File: public/app/features/alerting/unified/components/rules/RulesGroup.tsx [hunk 1]
```diff
@@ -209,9 +209,12 @@ export const RulesGroup = React.memo(({ group, namespace, expandAll, viewMode }:
             />
           </Tooltip>
         )}
-        <h6 className={styles.heading}>
-          {isFederated && <Badge color="purple" text="Federated" />} {groupName}
-        </h6>
+        {
+          // eslint-disable-next-line
+          <div className={styles.groupName} onClick={() => setIsCollapsed(!isCollapsed)}>
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rules/RulesGroup.tsx:214]: 💯

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM! 🎉
(Also tested locally and everything looks great!)

File: public/app/features/alerting/unified/components/rules/RulesGroup.tsx (hunk 1 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rules/RulesGroup.tsx:214]: yes, totally agree 👌, thank you!

**Pull request merged**